lib/gamora/plugs/adapters/idp_adapter.ex

```diff
@@ -25,6 +26,13 @@ defmodule Gamora.Plugs.AuthenticatedUser.IdpAdapter do
     end
   end
 
+  defp get_current_user(%Conn{} = conn) do
+    case conn.assigns[:current_user] do
+      nil -> {:ok, :pending_user}
+      user -> {:ok, user}
```

Comment:
shoudl thsi be matched to a Gamora.User struct?

Reply:
I want to make it flexible. It always depends on the client. For example, with the Edna JWT token authorization the current user is a BS User because of the exiting process.

Reply:
The weird thing is that you are building a struct for pending user, so the call method will return 2 different data types, so IMO, then relax the struct above, or return the same struct here, but having 2 different data types returned from the same function seems weird,

Reply:
As we talked offline, this will be kept until the whole Amco ecosystem gets migrated to IDP.